docs/MIGRATION-1.8-to-1.9.rst

```diff
@@ -0,0 +1,205 @@
+
+Migrating from ODC 1.8.x to 1.9.x
+=================================
+
+The last new major release of the Open Data Cube was v1.8.0 in May 2020, nearly 4 years ago.
+
+ODC developers and the Steering Council have been working hard behind the scenes over the last couple of years
+to address some of the accumulated technical debt in datacube-core and prepare for new major releases.
+
+The long-term plan includes a number of significant backwards-incompatible changes.  An effort has been made to
+provide a smooth migration pathway wherever possible, with existing behaviour in 1.8.x versions being deprecated
+in 1.9.x, with alternatives being provided, then removing the deprecated behaviour in 2.0.x with the alternative
+approaches becoming the standard, but some minor backwards incompatible changes in 1.9.x were unavoidable.
+
+This document describes the changes between 1.8.x and 1.9.x, with a particular focus on backwards incompatible
+changes and new features.
+
+After  the release of 1.9.0, focus will shift to updating secondary ODC libraries to work with ODC-1.9.  (Explorer
+and OWS in particular will require major changes.)   We will continue to support and release 1.8.x versions after
+the release of 1.9.0, until the 1.9.x releases have stabilised and all secondary libraries are up to date.
+
+Smaller ODC installations will probably prefer to stick with the 1.8.x releases for the time being, but if you can
+spare the resources we encourage you to set up a 1.9.x installation to test your existing code and systems
+against the new release, and open issues on github for any problems that you come up against, especially any that are
+not documented here.
+
+Major Changes between 1.8.x and 1.9.x
+-------------------------------------
+
+1. Integration with `odc-geo`.
```

Comment:
If version 1.9 will add the new `odc-geo` `.odc.<>` accessor to data loaded from datacube (it should if it doesn't), we should mention it here - access to `odc-geo` tools via that accessor will be hugely impactful for our users. Ideally, we want users to stop using the older methods/properties in favour of the `odc-geo` version (e.g. use `ds.odc.geobox` instead of `ds.geobox`)

Reply:
The `.odc` extension is automagically added to xarray DataArrays and Datasets (i.e. after loading) - and yes is worth mentioning.  

Adding it to `datacube.model.Dataset` (i.e. before loading) would require manual work which has not been done.  Note that only a small subset of the `.odc` accessor would make sense on an `datacube.model.Dataset` object which represents metadata rather than a raster of actual data.  Definitely something to think about though - maybe open an issue?